cmds/webboot/types.go

```diff
@@ -1,32 +1,44 @@
 package main
 
-import ui "github.com/gizak/termui/v3"
+import (
+	ui "github.com/gizak/termui/v3"
+	"github.com/u-root/webboot/pkg/menu"
+)
 
 const (
 	tcURL    = "http://tinycorelinux.net/10.x/x86_64/release/TinyCorePure64-10.1.iso"
 	wbtcpURL = "https://github.com/u-root/webboot-distro/raw/master/iso/tinycore/10.x/x86_64/release/TinyCorePure64.iso"
 )
 
 // bookmark record the list of iso webboot allow user to download
-var bookmark = []*BookMarkISO{
-	&BookMarkISO{
-		url:   tcURL,
-		label: "Download Tinycore v10.1",
-		name:  "TinyCorePure64-10.1.iso",
-	},
-	&BookMarkISO{
-		url:   wbtcpURL,
-		label: "Webboot Tinycorepure",
-		name:  "TinyCorePure64.iso",
-	},
-}
+var (
+	bookmarks = []menu.Entry{
+		&BookmarkGroup{
+			label: "Tinycore",
+			entries: []menu.Entry{
+				&BookmarkGroup{
+					label: "TinyCorePure64",
+					entries: []menu.Entry{
+						&BookMarkISO{
+							url:   tcURL,
+							label: "TinyCorePure64-10.1.iso",
+						},
+						&BookMarkISO{
+							url:   wbtcpURL,
+							label: "Webboot_Tinycorepure.iso",
+						},
+					},
+				},
+			},
+		},
+	}
+)
 
 // BookMarkISO contains information of the iso user want to download
 // expected to implement Entry interface
 type BookMarkISO struct {
```

Comment:
I'm wondering if this has to be a different struct from CachedISO; is there a way it doesn't have to be?

Reply:
Yes, BookMarkISO's exec will download the iso first then boot and the cachedISO will just boot it.